### ISSUE TITLE
Edit or View Metadata freezes browser 

### ISSUE THREAD

**Issue by yarikoptic** [dandi/dandiarchive#835]
addition to a generic #662 

I could trigger only on staging ATM and on some (but not all, if I got my ownership right) dandisets

If I login and go to https://gui-staging.dandiarchive.org/#/dandiset/000020 , and hit "Edit Metadata" - it freezes my chromium tab.  Eventually browser says that page is non-responsive and then Aw, Snap.  I reloaded, logged out, clicked "View Metadata" and got the same effect.

I have tried in brave browser (not logged in) to the same effect.

FTR:
- Version of web ui reported was 0.1.0-586-g52a0d35 
- https://github.com/dandi/dandi-api-webshots/ does not report oddities on deployed instance/dandisets with version only 2 commits before (0.1.0-584-g95039f3) with a diff which unlikely relevant, so most likely it is this staging dandiset metadata which brings it down? (thus not as critical issue as it could be if happened in production)

**Comment by yarikoptic** [comment]
FWIW, doesn't happen for any preceding dandiset (from 000001) and crashes on 000020 (discovered while preparing https://github.com/dandi/dandi-api-webshots-staging )

**Comment by waxlamp** [comment]
@mvandenburgh and I did some investigation just now.

We figured out that migrating 000020's metadata to the latest schema, followed by manually having the server revalidate it, "solves" the problem of freezing the browser, but we don't know why. Mike is continuing to investigate how the metadata is crashing the browser in his local setup. But for now, @yarikoptic, you should no longer have any trouble with 000020. If you run into a similar problem with a different dandiset, please let us know about it here.

Leaving this open with Mike investigates--he will report his findings here as well.

**Comment by yarikoptic** [comment]
didn't test "first hand" but next attempt to run https://github.com/dandi/dandi-api-webshots-staging finished with
```
2021-08-10T15:08:48-0400 [INFO    ] 000021 landing
2021-08-10T15:08:50-0400 [INFO    ] 000021 edit-metadata
Traceback (most recent call last):
  File "tools/make_webshots.py", line 277, in <module>
    stats = process_dandiset(driver, ds)
  File "tools/make_webshots.py", line 184, in process_dandiset
    act()
  File "tools/make_webshots.py", line 146, in click_edit
    edit_button.click()
  File "/home/dandi/cronlib/dandi-api-webshots-staging/venv/lib/python3.8/site-packages/selenium/webdriver/remote/webelement.py", line 80, in click
    self._execute(Command.CLICK_ELEMENT)
  File "/home/dandi/cronlib/dandi-api-webshots-staging/venv/lib/python3.8/site-packages/selenium/webdriver/remote/webelement.py", line 633, in _execute
    return self._parent.execute(command, params)
  File "/home/dandi/cronlib/dandi-api-webshots-staging/venv/lib/python3.8/site-packages/selenium/webdriver/remote/webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "/home/dandi/cronlib/dandi-api-webshots-staging/venv/lib/python3.8/site-packages/selenium/webdriver/remote/errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: unknown error: session deleted because of page crash
from unknown error: cannot determine loading status
from tab crashed
  (Session info: headless chrome=73.0.3683.75)
```
so I guess that some series of datasets which were created at some point would trigger this issue.

**Comment by mvandenburgh** [comment]
I figured out the source of the freezing. The `roleName` field was using the old `dandi:ContactPerson` syntax that hasn't been used since schema version 3.x, but the `schemaVersion` field said the metadata was on schema version 0.4.4 which uses the `dcite:ContactPerson` syntax. Changing the `roleName` field to use the 0.4.4 syntax fixes the freezing.

Not sure how this happened, maybe a schema migration bug? It's concerning that vjsf freezes up completely when there's a malformed schema field :frowning_face:

**Comment by waxlamp** [comment]
If you can cook up a repro to send upstream to VJSF, that would be cool.

Otherwise, can we easily fix the problem on all existing dandisets? If we can, then I think that would be the cue to close this issue.

And finally, is there something we can file against dandischema? Probably not at this time, I'm guessing. If this happens again, at least we'll have one new place to look immediately.

**Comment by mvandenburgh** [comment]
Update: this problem was fixed in vjsf 2.3.1.

**Comment by yarikoptic** [comment]
Issue filed/fixed: https://github.com/koumoul-dev/vuetify-jsonschema-form/issues/289
